FUSETOOLS2-115: Include UI tests for WSDL2REST into PR checks

mlorinc opened the pull request.

apupier commented on src/ui-test/extension.test.ts:
```diff
@@ -0,0 +1,416 @@
+}
+
+async function getInput(driver: WebDriver): Promise<Input> {
+	// avoid sleep conditions
```
I think this comment can be removed, I guess the sleep conditions has been avoided

djelinek replied:
+1

apupier commented on .travis.yml:
```diff
@@ -18,13 +17,14 @@ install:
 cache:
   directories:
-    - "node_modules"
+    - "$HOME/.m2"
```
why the "node_modules" cache has been removed?

mlorinc replied:
I thought `opencv4nodejs-prebuilt` package was causing problems because it contains C++ addons. This is why I removed it from cache. However I have performed tests again with cached `node_modules` . Tests passed, so I am going to revert this change.

apupier commented on .vscodeignore:
```diff
@@ -8,4 +10,5 @@ sonar-project.properties
 tsconfig.json
```
as "**/tsconfig.json" is added, I think this line can be removed

mlorinc replied:
I have also added `.ui-testing` and `out/ui-test/**` to `.vscodeignore`.

apupier commented on package.json:
```diff
@@ -84,7 +86,9 @@
         "typescript": "^3.7.5",
-        "vscode-test": "^1.3.0"
+        "vscode-test": "^1.3.0",
+        "vscode-extension-tester": "^2.4.0",
```
it is better to try to keep the list of dependencies in alphabetical order. it is easier to find one when reading

apupier commented on src/ui-test/extension.all.test.ts:
```diff
@@ -0,0 +1,81 @@
+import * as marketplaceTest from './marketplace.test';
+import * as fs from 'fs';
```
not mandatory: usually better to try to have import in alphabetical order

apupier commented on src/ui-test/runtimes/spring/README.md:
```diff
@@ -0,0 +1,26 @@
+3. Run command from command line `mvn -Dcamel.version=2.24.2 exec:java`
+
+
+## Buld steps
```
typo "Buld" -> "Build"

apupier commented on src/ui-test/test-data/fuse/blueprint-url-2.21.0.fuse-730078-redhat-00001.json:
```diff
@@ -0,0 +1,5 @@
+{
+    "framework": "blueprint",
+    "type": "url",
+    "camelVersion": "2.21.0.fuse-730078-redhat-00001"
```
maybe we should use the latest version available which should be a 7.5.0. What do you think?

mlorinc replied:
Thank you for pointing out. I will definitely change it. It should be 2.21.0.fuse-750033-redhat-00001, should not it?

apupier commented on src/ui-test/test-data/community/spring-url-2.24.2.json:
```diff
@@ -0,0 +1,5 @@
+{
+    "framework": "spring",
+    "type": "url",
+    "camelVersion": "2.24.2"
```
maybe we should use the latest 2.x version? what do you think?

mlorinc replied:
Agree, changed version to 2.25.0.

djelinek replied:
reflect this version change also in all readme files please

djelinek commented on src/ui-test/runtimes/blueprint/README.md:
```diff
@@ -0,0 +1,26 @@
+## Steps using wsdl2rest vscode extension
+
+1. Create or find wsdl file. It can be file in filesystem or url to file.
+2. Run the vscode extesions command.
```
typo: "extesions" -> "extensions"?

djelinek replied:
typo: "extesions" -> "extensions"?

djelinek commented on src/ui-test/runtimes/spring/README.md:
```diff
@@ -0,0 +1,26 @@
+## Steps using wsdl2rest vscode extension
+
+1. Create or find wsdl file. It can be file in filesystem or url to file.
+2. Run the vscode extesions command.
```
typo: "extesions" -> "extensions"?

mlorinc replied:
Simplified it to `2. Run extension command.`

djelinek commented on src/ui-test/package_data.ts:
```diff
@@ -0,0 +1,41 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as assert from 'assert';
```
not mandatory: imports in alphabetical order

djelinek commented:
Hi @apupier, regarding to your question
> * ui-test are not played on Jenkins, is it planned in a separate PR?

I noticed UI tests are not played on Jenkins for each of our extensions, so we need to update project Jenkinsfile everywhere, I created new [issue](https://issues.redhat.com/browse/FUSETOOLS2-234) for each component inside Jira project and it will be provided by another PR(s) across all extensions. Is this approach suitable? 

Thanks for raising this!

mlorinc commented:
First of all, I would like to thank you for your valuable feedback and apologize for my late response. 

### Changelog

* added license headers
* updated Camel versions (2.25.0 and 2.21.0.fuse-750033-redhat-00001)
* generated new runtimes from [generator-camel-project](https://github.com/camel-tooling/generator-camel-project). Differences in pom are commented with `wsdl2rest depencencies`. Also pom contains extra `camel-maven-plugin` for standalone support.
* when testing runtime, extra property `camel.maven.plugin.version` is passed. It is workaround when camel 2.25.0 would not work with camel-maven-plugin 2.25.0.
* changed runtime readme files
* sorted imports
* new commits FUSETOOLS2-257 and FUSETOOLS2-250 (fixes)

mlorinc commented:
The following error occurs when directory `.ui-testing` already exists. If it does not, please let me know. Maybe I should add better error message.

mlorinc commented on src/ui-test/extension.all.test.ts:
```diff
@@ -0,0 +1,102 @@
+
+		if (stat && stat.isDirectory()) {
+			// add directories and files to stack and transform filenames to absolute paths
+			stack.push(...fs.readdirSync(file).map(f => path.join(file, f)));
```
It converts string array to varargs.

apupier replied:
thanks for the explanation. I learned something today :-)

apupier commented:
> The following error occurs when directory `.ui-testing` already exists. If it does not, please let me know. Maybe I should add better error message.

it was that.
Please add a better error message

mlorinc commented on src/ui-test/extension.test.ts:
```diff
@@ -0,0 +1,445 @@
+
+			await input.test({
+				placeholder: 'Specify which DSL to generate the Camel configuration for',
+				// quickPicks: ['Spring', 'Blueprint']
```
This part is not commented in Update 2. After updating vscode-extension-tester it started working again.

Related issue: [https://issues.redhat.com/browse/FUSETOOLS2-257](https://issues.redhat.com/browse/FUSETOOLS2-257)

djelinek replied:
Could you please add note about related issue to the commented line into code ?

mlorinc commented on src/ui-test/extension.all.test.ts:
```diff
@@ -0,0 +1,102 @@
+	expect(project.exists).to.be.true;
+	
+	await project.open();
+	await DefaultWait.sleep(12000);
```
I could try wait for title bar to show folder name. I am going to try it.

djelinek replied:
I would recommend to let it here for now to be able to move on and fix it with another PR, after issue [FUSETOOLS2-317](https://issues.redhat.com/browse/FUSETOOLS2-317) will be resolved. Because it is affecting more components.. WDYT? @apupier @mlorinc

mlorinc replied:
In my point of view, we should use sleep for now. New wait condition partially works but it occasionally fails.

apupier replied:
ok

apupier commented:
@mlorinc can you rebase and resolved conflict in package-lock.json please?
